rpc/dialer.go

```diff
@@ -307,14 +307,15 @@ func dialDirectGRPC(ctx context.Context, address string, dOpts *dialOptions, log
 			// added as well as an authorization header added from perRPCJWTCredentials, resulting in a failure.
 			externalAuthMaterial: dOpts.externalAuthMaterial,
 		}
-		if dOpts.debug {
+		if dOpts.debug && dOpts.externalAuthAddr != "" && dOpts.externalAuthToEntity != "" {
 			logger.Debugw("will eventually authenticate as entity", "entity", dOpts.authEntity)
 		}
 		if dOpts.externalAuthAddr != "" {
-			if dOpts.debug && dOpts.externalAuthToEntity != "" {
-				logger.Debugw("will eventually externally authenticate to entity", "entity", dOpts.externalAuthToEntity)
+			if dOpts.externalAuthToEntity == "" {
+				return nil, false, errors.New("external auth address set but no authenticate to option set")
```

Comment:
This is new. We've never used this particular combination of settings and I don't see how it's useful to use your externally got credentials without using authTo. Felt like a bug to me but I had tests written to support it.

Reply:
[q] So our customers always have to AuthTo a user in their org in order to use the API?

Reply:
Nope, just to control/use to a robot if they are passing JWTs to the robot directly (not using our webrtc signaling in the cloud)